File: bit_manipulation/0338_couting_bits/0338_couting_bits.py
class Solution(object):
    def countBits(self, num):
        res = [0]
        for i in xrange(1, num + 1):
            res.append((i&1)+res[i >> 1])
        return res

    def countBits2(self, num):
        res = [0]
        for i in xrange(1,num+1):
            r = res[i&(i-1)] + 1
            res.append(r)

        return res

num = 5
res = Solution().countBits2(num)
print(res)

File: bit_manipulation/0338_couting_bits/0338_couting_bits.go
package main 

import "fmt"

func countBits(num int) []int {
	res := make([]int, num +1)
	for i := 1; i <= num; i++ {
		res[i] = res[i>>1] + i & 1
	}

	return res
}

func main() {
	num := 5
	res := countBits(num)
	fmt.Println(res)
}